operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -84,4 +84,24 @@ protected Integer currentScale(String namespace, String name) {
         Annotations.annotations(desired).put(Annotations.ANNO_DEP_KUBE_IO_REVISION, k8sRev);
         return super.internalPatch(namespace, name, current, desired, cascading);
     }
+
+    public Future<Void> generation(String namespace, String name, long pollIntervalMs, long timeoutMs) {
```

Comment:
Add Javadoc please.

We should also here find some better name. We are trying to find out if the deployment has been updated. So maybe something based on that?

Reply:
Would `observing` be a good replacement for `generation`? Or would that be too general?

Reply:
I was of some other names like `updatedDeployment`, `latestDeployment`, `observingDeploymentGeneration`